ignite3/src/main/java/site/ycsb/db/ignite3/IgniteJdbcClient.java

```diff
@@ -215,4 +173,88 @@ public void cleanup() throws DBException {
 
     super.cleanup();
   }
+
+  /**
+   * Perform single INSERT operation with Ignite JDBC client.
+   *
+   * @param key Key.
+   * @param values Values.
+   */
+  protected void put(String key, Map<String, ByteIterator> values) throws SQLException {
+    PreparedStatement stmt = INSERT_PREPARED_STATEMENT.get();
+
+    setStatementValues(stmt, key, values);
+
+    stmt.executeUpdate();
+  }
+
+  /**
+   * Perform single SELECT operation with Ignite JDBC client.
+   *
+   * @param key Key.
+   * @param fields Fields.
+   * @param result Result.
+   */
+  @NotNull
+  protected Status get(String key, Set<String> fields, Map<String, ByteIterator> result) throws SQLException {
+    PreparedStatement stmt = READ_PREPARED_STATEMENT.get();
+
+    stmt.setString(1, key);
+
+    try (ResultSet rs = stmt.executeQuery()) {
+      if (!rs.next()) {
+        return Status.NOT_FOUND;
+      }
+
+      if (fields == null || fields.isEmpty()) {
+        fields = new HashSet<>();
+        fields.addAll(this.valueFields);
+      }
+
+      for (String column : fields) {
+        //+2 because indexes start from 1 and 1st one is key field
+        String val = rs.getString(this.valueFields.indexOf(column) + 2);
+
+        if (val != null) {
+          result.put(column, new StringByteIterator(val));
+        }
+      }
+    }
+
+    return Status.OK;
+  }
+
+  /**
+   * Perform single UPDATE operation with Ignite JDBC client.
+   *
+   * @param key Key.
+   * @param values Values.
+   */
+  private void modify(String key, Map<String, ByteIterator> values) throws SQLException {
+    try (Statement stmt = CONN.get().createStatement()) {
+      List<String> updateValuesList = new ArrayList<>();
+
+      for (Entry<String, ByteIterator> entry : values.entrySet()) {
+        updateValuesList.add(String.format("%s='%s'", entry.getKey(), entry.getValue().toString()));
+      }
+
+      String sql = String.format("UPDATE %s SET %s WHERE %s = '%s'",
+          cacheName, String.join(", ", updateValuesList), PRIMARY_COLUMN_NAME, key);
```

Comment:
this is out of scope of current patch, still I want to highlight this will never have decent performance. First problem is inlining value for `key`, we always must use dynamic parameter `?` in such a case. Second problem is the way we generate `SET` part of the statement: `updateValuesList` is populated in the order values appear in `value` map. Given that map doesn't preserve order, it potentially creates `values.size()!` combinations.

Reply:
Yes, this place is left unchanged during the code extraction.
Actually, for now we don't use the `update` operation in our benchmark scenarios, but I will create a ticket to refactor this.

Reply:
https://ggsystems.atlassian.net/browse/QA-5627